java/ql/src/experimental/semmle/code/java/security/DecompressionBomb.qll

```diff
@@ -0,0 +1,390 @@
+import java
+private import semmle.code.java.dataflow.TaintTracking
+
+module DecompressionBomb {
+  /**
+   * The Decompression bomb Sink
+   *
+   * Extend this class for creating new decompression bomb sinks
+   */
+  abstract class Sink extends DataFlow::Node { }
+
+  /**
+   * The Additional flow steps that help to create a dataflow or taint tracking query
+   *
+   * Extend this class for creating new additional taint steps
+   */
+  class AdditionalStep extends Unit {
+    abstract predicate step(DataFlow::Node n1, DataFlow::Node n2);
+  }
+
+  abstract class BombReadInputStreamCall extends MethodCall { }
+
+  private class ReadInputStreamQualifierSink extends DecompressionBomb::Sink {
+    ReadInputStreamQualifierSink() { this.asExpr() = any(BombReadInputStreamCall r).getQualifier() }
+  }
+
+  abstract class BombTypeInputStream extends RefType { }
```

Comment:
```suggestion

  abstract class BombTypeInputStream extends RefType { }
```
Now that you've removed `TypeInputStreamConstructorArgumentSink` there is no use for this class any more. Please delete it and the classes that extend it can go back to extending `RefType` as before.